packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -1089,21 +1111,122 @@ class MockDocument {
   }
 }
 
+/**
+ * `WebStorage` mock that implements the WebStorage API used by Firestore.
+ */
+class MockStorage {
+  private static sharedData = new Map<string, string>();
+  private static activeInstances = new Map<EventListener, MockStorage>();
+
+  eventListener: EventListener | null = null;
```

Comment:
So the data is static but the listener is not?  But there's a static map of listeners to MockStorage instances (where the only non-static member is the eventListener)?  This seems funky to me.  I don't know how I would describe this class. It's like a handle to a shared Storage instance but with a non-shared listener or something.

Would it make sense to drop `activeInstances` and just have a static array of `listeners`? and change start() / shutdown() into addListener() / removeListener() or similar?

That said I kinda' expect test mocks not to have any static state since that could bleed over between tests.  So maybe nothing should be static and you should just make sure that the same instance is used throughout the test (which may already be happening?  I'm not sure.)